PYME/Acquire/protocol.py

```diff
@@ -192,11 +192,14 @@ def __init__(self, taskList, startFrame, dwellTime, metadataEntries=[], prefligh
             self.slice_order = slice_order
 
         self.require_camera_restart = require_camera_restart
-
-    def Init(self, spooler):
-        self.zPoss = np.arange(scope.stackSettings.GetStartPos(),
-                               scope.stackSettings.GetEndPos() + .95 * scope.stackSettings.GetStepSize(),
-                               scope.stackSettings.GetStepSize() * scope.stackSettings.GetDirection())
+    
+    def set_stack_positions(self, stack_mdh=None):
```

Comment:
a) see PR #799 for re-factored `StackSettings` object. Any function to set one of these should either take a `StackSettings` object, or a dict of arguments (as provided by the http endpoints), **not** a metadata handler.  

A dedicated object/handler is needed as some of the properties are computed rather than directly read, depending on stack mode (hard ends, or middle & number).  It makes sense that the logic for computing these is in one place and not duplicated. Notably, this code will probably not do what you want as the default mode is middle & number. Protocol then uses it's own `StackSettings` object/the one that it gets passed rather than the scope one, leaving the rest of the code the same.

a) having this as a separate function (rather than in `Init()`) feels a bit off, but could grow on me.